monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js

```diff
@@ -366,110 +324,65 @@ class ConfigurePageComponent extends AuthComponent {
-    }];
-    pbaFile[0].options.file = mockFile;
-    return pbaFile
-  }
-
   renderMatrix = () => {
     return (<ConfigMatrixComponent configuration={this.state.attackConfig}
                                    submit={this.componentDidMount}
                                    reset={this.resetConfig}
                                    change={this.attackTechniqueChange}/>)
   };
 
-
   renderConfigContent = (displayedSchema) => {
-    return (<div>
-      {this.renderBasicNetworkWarning()}
-      <Form schema={displayedSchema}
-            uiSchema={this.uiSchemas[this.state.selectedSection]}
-            formData={this.state.configuration[this.state.selectedSection]}
-            onChange={this.onChange}
-            noValidate={true}
-            className={'config-form'}>
-        <button type='submit' className={'hidden'}>Submit</button>
-      </Form>
-    </div>)
+    let formProperties = {};
+    formProperties['schema'] = displayedSchema
+    formProperties['uiSchema'] = UiSchema({
+      PBA_linux_filename: this.state.configuration.monkey.post_breach.PBA_linux_filename,
+      PBA_windows_filename: this.state.configuration.monkey.post_breach.PBA_windows_filename,
+      setPbaFilenameWindows: this.setPbaFilenameWindows,
+      setPbaFilenameLinux: this.setPbaFilenameLinux,
+      selectedSection: this.state.selectedSection
+    })
+    formProperties['formData'] = this.state.configuration[this.state.selectedSection]
+    formProperties['onChange'] = this.onChange
+    formProperties['customFormats'] = formValidationFormats
+    formProperties['className'] = 'config-form'
+    formProperties['liveValidate'] = true
+
+    if (this.state.selectedSection === 'internal') {
+      return (<InternalConfig {...formProperties}/>)
+    } else {
+      return (
+        <div>
+          {this.renderBasicNetworkWarning()}
+          <Form {...formProperties}>
+            <button type='submit' className={'hidden'}>Submit</button>
+          </Form>
+        </div>
+      )
+    }
   };
 
+  setPbaFilenameWindows = (filename) => {
+    let config = this.state.configuration
+    config.monkey.post_breach.PBA_windows_filename = filename
+    this.setState({
+      configuration: config
+    })
+  }
+
+  setPbaFilenameLinux = (filename) => {
+    let config = this.state.configuration
+    config.monkey.post_breach.PBA_linux_filename = filename
+    this.setState({
+      configuration: config
+    })
+  }
+
   renderBasicNetworkWarning = () => {
     if (this.state.selectedSection === 'basic_network') {
       return (<div className='alert alert-info'>
         <FontAwesomeIcon icon={faInfoCircle} style={{'marginRight': '5px'}}/>
-        The Monkey scans its subnet if 'Local network scan' is ticked. Additionally the monkey scans machines
-        according to its range class.
+        The Monkey scans its subnet if "Local network scan" is ticked. Additionally the monkey scans machines
+        according to "Scan target list".
       </div>)
```

Comment:
We should move this warning into the "scope" section of the configuration page. It's out of place at the top of the page.